test/extended/networking/network_segmentation.go

```diff
@@ -682,16 +706,295 @@ var _ = Describe("[sig-network][OCPFeatureGate:NetworkSegmentation][Feature:User
 			expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 			Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 		})
+
+		Context("ClusterUserDefinedNetwork CRD Controller", func() {
+			const clusterUserDefinedNetworkResource = "clusteruserdefinednetwork"
+
+			var testTenantNamespaces []string
+
+			BeforeEach(func() {
+				testTenantNamespaces = []string{
+					f.Namespace.Name + "blue",
+					f.Namespace.Name + "red",
+				}
+
+				By("Creating test tenants namespaces")
+				for _, nsName := range testTenantNamespaces {
+					_, err := cs.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					DeferCleanup(func() error {
+						err := cs.CoreV1().Namespaces().Delete(context.Background(), nsName, metav1.DeleteOptions{})
+						return err
+					})
+				}
+			})
+
+			var testClusterUdnName string
+
+			BeforeEach(func() {
+				testClusterUdnName = randomNetworkMetaName()
+				By("create test CR")
+				cleanup, err := createManifest("", newClusterUDNManifest(testClusterUdnName, testTenantNamespaces...))
+				DeferCleanup(func() error {
+					cleanup()
+					_, _ = e2ekubectl.RunKubectl("", "delete", clusterUserDefinedNetworkResource, testClusterUdnName)
+					Eventually(func() bool {
+						_, err := e2ekubectl.RunKubectl("", "get", clusterUserDefinedNetworkResource, testClusterUdnName)
+						return strings.Contains(err.Error(), fmt.Sprintf("clusteruserdefinednetworks.k8s.ovn.org %q not found", testClusterUdnName))
+					}, 1*time.Minute, 3*time.Second).Should(BeTrueBecause("ClusterUserDefinedNetwork %q should be gone", testClusterUdnName))
```

Comment:
shouldn't we match error instead of this ? 

It's so much more readable ...

Reply:
I wish we could, the `e2ekubectl.RunKubectl` doesnt play nice with the error matcher, 
because `e2ekubectl.RunKubectl` return some custom error in a formatting that makes it impossible to convert to some standard k8s error.

Reply:
OK, I had just glanced, and missed the fact there's actually a reason to treat the error as a string ...